### PR TITLE
Replace V1 annotations with V2 annotations (PLFM-5727)

### PR DESCRIPTION
**Pull request by zimingd**

### CODE REVIEW

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImpl.java [hunk 1]
```diff
@@ -734,6 +730,34 @@ private Annotations getAnnotations(final String id, Long version, final String a
...
+		if(jsonString != null){
+			try {
+				annos = EntityFactory.createEntityFromJSONString(jsonString, AnnotationsV2.class);
+			} catch (JSONObjectAdapterException e) {
+				throw new DatastoreException(e);
+			}
+		}else{
+			// If there is no annotations then create a new one.
+			annos = new AnnotationsV2();
+			annos.setAnnotations(new HashMap<>());
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImpl.java]: duplicate

File: lib/models/src/main/java/org/sagebionetworks/repo/model/annotation/v2/AnnotationV1AndV2TypeMapping.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package org.sagebionetworks.repo.model.annotation.v2;
+
```
Comment (john-hill) [inline lib/models/src/main/java/org/sagebionetworks/repo/model/annotation/v2/AnnotationV1AndV2TypeMapping.java]: *annotations

File: lib/models/src/test/java/org/sagebionetworks/repo/model/annotation/v2/AnnotationsV2UtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+	@Test
+	public void toJSONStringForStorage_EmptyAnnotationsMap() throws JSONObjectAdapterException {
+		AnnotationsV2 annotationsV2 = new AnnotationsV2();
+		annotationsV2.setAnnotations(Collections.emptyMap());
+		assertNull(AnnotationsV2Utils.toJSONStringForStorage(annotationsV2));
+	}
+
+	@Test
+	public void toJSONStringForStorage_AnnotationsMapWithEntries() throws JSONObjectAdapterException {
+		AnnotationsV2 annotationsV2 = new AnnotationsV2();
```
Comment (john-hill) [inline lib/models/src/test/java/org/sagebionetworks/repo/model/annotation/v2/AnnotationsV2UtilsTest.java:124]: set ID and etag to confirm they do not make it to the json

File: lib/models/src/main/java/org/sagebionetworks/repo/model/annotation/v2/annotaitonvalidator/AnnotationsV2ValueValidator.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package org.sagebionetworks.repo.model.annotation.v2.annotaitonvalidator;
+
+public interface AnnotationsV2ValueValidator {
+	/**
+	 * true if value is valid. false otherwise.
+	 * @param value
+	 * @return true if value is valid. false otherwise.
```
Comment (john-hill) [inline lib/models/src/main/java/org/sagebionetworks/repo/model/annotation/v2/annotaitonvalidator/AnnotationsV2ValueValidator.java]: validate with impls throwing exceptions.

File: lib/models/src/main/java/org/sagebionetworks/repo/model/table/AnnotationType.java [hunk 5]
```diff
@@ -12,18 +14,20 @@
  */
 public enum AnnotationType{
 	
-	STRING	(new StringParser(), ColumnType.STRING),
-	LONG	(new LongParser(), ColumnType.INTEGER),
-	DOUBLE	(new DoubleParser(), ColumnType.DOUBLE),
-	DATE	(new DateParser(), ColumnType.DATE);
+	STRING	(new StringParser(), ColumnType.STRING, AnnotationsV2ValueType.STRING),
+	LONG	(new LongParser(), ColumnType.INTEGER, AnnotationsV2ValueType.LONG),
```
Comment (john-hill) [inline lib/models/src/main/java/org/sagebionetworks/repo/model/table/AnnotationType.java:18]: revert

Comment (zimingd) [reply lib/models/src/main/java/org/sagebionetworks/repo/model/table/AnnotationType.java:18]: This is needed for creating AnnotationDTO from AnnotationV2

**Review by john-hill — APPROVED** [review]

**Pull request merged**